munit-cats-effect/shared/src/main/scala/munit/CatsEffectFunFixtures.scala

```diff
@@ -0,0 +1,55 @@
+package munit
+
+import cats.effect.{IO, Resource}
+import cats.syntax.flatMap._
+
+import scala.concurrent.Promise
+
+trait CatsEffectFunFixtures extends FunFixtures { self: CatsEffectSuite =>
```

Comment:
It would be nice if we could somehow support suite-local and test-local resources via `Fixture[T]`. The problem is that `Fixture[T]` doesn't support async before/after methods but I wonder if that is something we should consider adding to improve this use-case 🤔

I am concerned it would have to be a breaking change unless we introduce something like `AsyncFixture[T]`, but I would love to avoid having separate types for sync/async fixtures.

I personally prefer `Fixture[T]` over `FunFixture[T]` because 1) it's easier to compose them when you have many and 2) they require less boilerplate per test case. It's also useful to have suite-local resources like a database connection.

This doesn't have to be blocking for this PR, I'm just thinking out loud here. We could go ahead and merge this change for now and add `Fixture[T]` support separately.